lib/presenter.rb

```diff
@@ -0,0 +1,45 @@
+class Presenter
+  def initialize(displayer:, clear:)
+    @displayer = displayer
+    @clear = clear
+  end
+
+  def present(board:, message:)
+    [clear, format_board(board), message].each do |render|
+      displayer.display(render)
+    end
+  end
+
+  private
```

Comment:
The private methods in this class seem fairly console-specific, which undermines abstracting out `Presenter`.

Reply:
Would you suggest injecting something into presenter (like a formatter) that will provide this functionality? If not do you know of any good options to format the board before sending it to be displayed?

Reply:
Dependency injection sounds like a good approach to me!

Reply:
@hnlee One more question. If I use dependency injection to pass in a formatter to presenter, should presenter pass the board object to formatter and formatter should flatten and format it? (does this violate the law of demeter?) Or should presenter have the private flatten method and pass the flattened board array to formatter? Or is there another approach I am missing?

Reply:
Good question! Given this situation, I would think about what the respective single responsibilities of presenter and formatter are to try to resolve where the method belongs.

Reply:
I believe it would be the formatters responsibility to prepare the board for the presenter which would place the flatten method in formatter. I believe this makes sense. My only concern is that passing the board through presenter in this manner feels smelly. I am not entirely sure why. Do you have any thoughts on presenter passing the board object through?

Reply:
@hnlee For now, I think I will pass the board object to formatter because I think it is not the presenter's responsibility. Thank you very much for all of your help!

Reply:
Sorry, just saw this comment.  Your reasoning sounds convincing to me!  👍